### PR TITLE
Wismy/x7 access

### PR DESCRIPTION
**Pull request by bsongis**

### CODE REVIEW

File: radio/src/datastructs.h [hunk 1]
```diff
@@ -919,6 +919,9 @@ static inline void check_struct()
 #elif defined(PCBXLITE)
   CHKSIZE(RadioData, 858);
   CHKSIZE(ModelData, 6157);
+#elif defined(PCBX7ACCESS)
+  CHKSIZE(RadioData, 861);
```
Comment (bsongis) [inline radio/src/datastructs.h]: then we need to have a different FourCC

File: radio/src/lua/CMakeLists.txt [hunk 2]
```diff
@@ -16,15 +16,16 @@ set(LUA_INCLUDES
...
-  add_lua_export_target(x9d     ${LUA_INCLUDES} -DPCBTARANIS -DPCBX9D -DPCBX9)
-  add_lua_export_target(x9e     ${LUA_INCLUDES} -DPCBTARANIS -DPCBX9E -DPCBX9)
-  add_lua_export_target(x10     ${LUA_INCLUDES} -DPCBHORUS   -DPCBX10)
-  add_lua_export_target(x12s    ${LUA_INCLUDES} -DPCBHORUS   -DPCBX12S)
-  add_lua_export_target(t12     ${LUA_INCLUDES} -DPCBTARANIS -DPCBX7 -DRADIO_T12)
-  add_lua_export_target(t16     ${LUA_INCLUDES} -DPCBHORUS   -DPCBX10 -DRADIO_T16)
+  add_lua_export_target(x9lite    ${LUA_INCLUDES} -DPCBTARANIS -DPCBX9LITE -DPCBX9)
+  add_lua_export_target(x9lites   ${LUA_INCLUDES} -DPCBTARANIS -DPCBX9LITES -DPCBX9LITE -DPCBX9)
+  add_lua_export_target(x7        ${LUA_INCLUDES} -DPCBTARANIS -DPCBX7  -DRADIO_X7)
+  add_lua_export_target(x7access  ${LUA_INCLUDES} -DPCBTARANIS -DPCBX7ACCESS -DPCBX7  -DRADIO_X7)
```
Comment (bsongis) [inline radio/src/lua/CMakeLists.txt]: one space too much between `-DPCBX7` and `-DRADIO_X7`

File: radio/src/datastructs.h (hunk 1 above)
Comment (bsongis) [reply radio/src/datastructs.h]: if possible I would prefer that both eeprom are compatible

File: radio/src/targets/common/arm/stm32/bluetooth_driver.cpp [hunk 3]
```diff
@@ -24,7 +24,9 @@
 Fifo<uint8_t, BT_TX_FIFO_SIZE> btTxFifo;
 Fifo<uint8_t, BT_RX_FIFO_SIZE> btRxFifo;
 
-#if defined(PCBX7) || defined(PCBXLITE)
+#if defined(PCBX7ACCESS)
```
Comment (bsongis) [inline radio/src/targets/common/arm/stm32/bluetooth_driver.cpp:27]: the code starts to be too much complex here, I propose we introduce:
- BLUETOOTH=YES
- BLUETOOTH=NO
- BLUETOOTH=PROBE

File: radio/src/targets/taranis/board.h [hunk 4]
```diff
@@ -396,6 +400,11 @@ enum EnumSwitchesPositions
   #define STORAGE_NUM_SWITCHES          NUM_SWITCHES
   #define DEFAULT_SWITCH_CONFIG         (SWITCH_2POS << 10) + (SWITCH_2POS << 8) + (SWITCH_3POS << 6) + (SWITCH_3POS << 4) + (SWITCH_3POS << 2) + (SWITCH_3POS << 0)
   #define DEFAULT_POTS_CONFIG           (POT_WITHOUT_DETENT << 0) + (POT_WITHOUT_DETENT << 2); // S1 = pot without detent, S2 = pot with detent
+#elif defined(PCBX7ACCESS)
+  #define NUM_SWITCHES                  7
+  #define STORAGE_NUM_SWITCHES          NUM_SWITCHES
```
Comment (bsongis) [inline radio/src/targets/taranis/board.h]: What about:
- NUMS_SWITCHES = 7
- STORAGE_NUM_SWITCHES = 8

File: radio/src/targets/taranis/lcd_driver_spi.cpp [hunk 5]
```diff
@@ -55,6 +55,26 @@ void lcdWriteCommand(uint8_t byte)
   LCD_NCS_HIGH();
 }
 
+
+void lcdWriteData(uint8_t* byte)
+{
+  LCD_A0_HIGH();
+  LCD_NCS_LOW();
+  for(uint8_t i=0;i<LCD_W;i++)
```
Comment (bsongis) [inline radio/src/targets/taranis/lcd_driver_spi.cpp]: code style not good

File: radio/src/targets/taranis/CMakeLists.txt [hunk 6]
```diff
@@ -97,22 +97,44 @@ elseif(PCB STREQUAL X9D)
...
   if(PCBREV STREQUAL T12)
+    set(CPU_TYPE STM32F2)
+    set(CPU_TYPE_FULL STM32F205xE)  # for size report
+    set(LINKER_SCRIPT targets/taranis/stm32f2_flash.ld)
     set(LUA_EXPORT lua_export_t12)
     set(FLAVOUR t12)
     set(NAVIGATION_TYPE 9x)
     add_definitions(-DRADIO_T12)
     add_definitions(-DEEPROM_VARIANT=0x4001)
-    option(INTERNAL_MODULE_MULTI "Support for MULTI internal module" OFF)
```
Comment (3djc) [inline radio/src/targets/taranis/CMakeLists.txt:110]: Why ?

File: radio/src/targets/taranis/lcd_driver_spi.cpp [hunk 7]
```diff
@@ -55,6 +55,26 @@ void lcdWriteCommand(uint8_t byte)
   LCD_NCS_HIGH();
 }
 
+
+void lcdWriteData(uint8_t* byte)
```
Comment (bsongis) [inline radio/src/targets/taranis/lcd_driver_spi.cpp]: the function name is not good

File: radio/src/targets/taranis/lcd_driver_spi.cpp [hunk 8]
```diff
@@ -205,6 +225,9 @@ void lcdRefresh(bool wait)
     lcdWriteCommand(0x04);
 #endif
 
+#if defined(PCBX7ACCESS)
```
Comment (bsongis) [inline radio/src/targets/taranis/lcd_driver_spi.cpp]: I would prefer we test if we have DMA on LCD, like this

`#if defined(LCD_DMA)`

File: radio/src/targets/taranis/CMakeLists.txt (hunk 6 above)
Comment (WismyYao) [reply radio/src/targets/taranis/CMakeLists.txt:110]: Sorry , my mistake .

File: radio/src/targets/taranis/lcd_driver_spi.cpp (hunk 8 above)
Comment (WismyYao) [reply radio/src/targets/taranis/lcd_driver_spi.cpp]: LCD has DMA but doesn't work : (

File: radio/src/targets/common/arm/stm32/bluetooth_driver.cpp (hunk 3 above)
Comment (bsongis) [reply radio/src/targets/common/arm/stm32/bluetooth_driver.cpp:27]: I will it myself later

**Pull request merged**